topsort-api-v2.yml

```diff
@@ -0,0 +1,603 @@
+        - Auctions
+      summary: Creates a new auction
+      operationId: createAuction
+      requestBody:
+        description: >
+          The information describing what will be auctioned.
+          Topsort will run an auction for each batched auction request, for which products, brands and/or vendors' bids will compete against each other.
+          Provided fields are additive, in the sense that only bids that match all filtering fields are considered (e.g. in the example below, only bids for products in the c_yogurt category that match the "Noosa Peach" search string will participate in the acution).
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                auctions:
+                  type: array
+                  items:
+                    oneOf:
+                      - $ref: '#/components/schemas/AuctionRequestSponsoredListings'
+                      - $ref: '#/components/schemas/AuctionRequestBannerAds'
+                    discriminator:
+                      propertyName: type
+                      mapping:
+                        listings: '#/components/schemas/AuctionRequestSponsoredListings'
+                        bannerAds: '#/components/schemas/AuctionRequestBannerAds'
+        required: true
+      responses:
+        201:
+          description: >
+            The auction results.
+            The list of winners will contain at most `slots` entries per auction.
+            It may contain fewer or no entries at all if there aren't enough products with usable bids.
+            Bid become unusable if campaign budget is exhausted, the bid is disqualified to preserve spend pacing, etc.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  results:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Auction'
+                example:
+                  results:
+                    - winners:
+                        - rank: 0
+                          winnerType: product
+                          winnerId: b_Mfk15
+                          resolvedBidId: b23f2d4b-ec5d-465f-b399-2344fc020c7
+                        - rank: 1
+                          winnerType: product
+                          winnerId: e_PJbnN
+                          resolvedBidId: 3aaa25fe-60a0-45a3-ae65-39ec1b525eb4
+                    - winners:
+                        - rank: 0
+                          winnerType: product
+                          winnerId: e_PJbnN
+                          resolvedBidId: 3aaa25fe-60a0-45a3-ae65-39ec1b525eb4
+                          assetUrl: https://address.to/lhs-banner-image-for-e_PJbnN.png
+        401:
+          $ref: '#/components/responses/UnauthorizedError'
+        400:
+          $ref: '#/components/responses/BadRequest'
+
+  /events:
```

Comment:
I wouldn't add the events endpoint yet. We need to reorganize that API, which has a lot of unused fields and it's not batchable.